### DISCUSSION TITLE
Local plugin not registering

### DISCUSSION THREAD

**Discussion by ali-ramadhan** [MoeMusic/Moe#227] [Q&A] [answered]
I'm trying to figure out how to use local plugins so following the documentation (https://mrmoe.readthedocs.io/en/latest/developers/writing_plugins.html#local-plugins).

I placed https://github.com/MoeMusic/moe_random/blob/main/moe_random/random.py under `<configuration directory>/plugins/random.py` and added `enable_plugins = ["random"]` to my configuration file but Moe doesn't seem to see it (or rather, I get no `random` command when I run `moe --help`).

```
(moe-py3.9) ➜  ~ exa -lT ~/.config/moe/
drwxr-xr-x    - alir  3 Nov 13:24 /home/alir/.config/moe
lrwxrwxrwx   40 alir  2 Nov 17:07 ├── config.toml -> /home/alir/music-tagging/moe_config.toml
.rw-r--r--  53k alir  3 Nov 13:24 ├── library.db
drwxr-xr-x    - alir  3 Nov 13:40 └── plugins
.rw-r--r-- 1.9k alir  2 Nov 19:36    └── random.py
```

My Python is pretty rusty so sorry if I'm doing something obviously wrong!

**Comment by ali-ramadhan** [comment]
I think I figured it out. I just had to add `~/.config/moe/plugins` to the `PYTHONPATH` environment variable.

Then the issue was that `random.py` clashed with the `random` module from the standard library so I renamed it to `random_item.py` and it worked.

Should a note about including the plugins directory in `PYTHONPATH` be added to the [documentation on local plugins](https://mrmoe.readthedocs.io/en/latest/developers/writing_plugins.html#local-plugins)?

**Comment by ali-ramadhan** [comment, marked as answer]
Hmmm never mind I think it works without `PYTHONPATH`, maybe the real issue was `random.py` silently conflicting with stdlib `random`.

**Reply by jtpavlock** [reply]
Nice catch, it does indeed seem that plugins with the same name as a module in the stdlib (or any other module in the namespace I would imagine) causes issues. I'll add a note in the docs to watch out for name conflicts. This should only effect local plugins though since published plugins should all be prefaced with `moe_`. 

Edit: added note in the docs 2e081261c696878bd25870393516c36850e19869
